src/metabase/api/user.clj

```diff
@@ -161,11 +161,24 @@
     user
     ((resolve 'metabase-enterprise.advanced-permissions.common/with-advanced-permissions) user)))
 
+(defn- add-has-question-and-dashboard
+  "True when the user has permissions for at least one un-archived question and one un-archived dashboard."
+  [user]
+  (let [coll-ids-filter (collection/visible-collection-ids->honeysql-filter-clause
+                          :collection_id
+                          (collection/permissions-set->visible-collection-ids @api/*current-user-permissions-set*))
+        perms-query {:where [:and
+                             [:= :archived false]
+                             coll-ids-filter]}]
+    (assoc user :has_question_and_dashboard (and (> (db/count 'Card (perms-query user)) 0)
```

Comment:
you can use `toucan.db/exists?` instead of a count > 0 here.

Reply:
also, this notion seems quite strange to me. If someone had 500 cards and 0 dashboards or someone had no permissions to see anything this would report the same. Seems weird to lump cards and dashboards together like this.